pdns/syncres.cc

```diff
@@ -4533,18 +4533,21 @@ void SyncRes::parseEDNSSubnetAddFor(const std::string& subnetlist)
 }
 
 // used by PowerDNSLua - note that this neglects to add the packet count & statistics back to pdns_recursor.cc
-int directResolve(const DNSName& qname, const QType qtype, const QClass qclass, vector<DNSRecord>& ret)
+int directResolve(const DNSName& qname, const QType qtype, const QClass qclass, vector<DNSRecord>& ret, shared_ptr<RecursorLua4> pdl)
```

Comment:
To me it always feels a bit wrong to pass "large" objects by value. You might want to change to a ref here.

Reply:
I agree in principle, but in that specific case taking a reference makes it a bit harder that simply passing `nullptr` when there is no Lua context, for example in `sendNODLookup()`. As the performance impact should be negligible in that case, I'd rather keep it as it is.